src/drivers/imx/esai.c

```diff
@@ -197,17 +197,23 @@ static inline int esai_set_config(struct dai *dai,
 
 	xccr |= ESAI_xCCR_xHCKD; /* Set the HCKT pin as an output */
 
-	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr); /* rx */
-	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr); /* tx */
+	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr);
+	/* We must make our rx clock slave because codec is master
+	 * TODO make this come from topology
+	 */
+	xccr &= ~(ESAI_xCCR_xCKD | ESAI_xCCR_xFSD);
```

Comment:
Lets add a proper explanation here: "There is a hardware limitation for using tx/rx in SAI
master. Workaround for this is to use tx master  / rx slave"